src/UCTNode.cpp

```diff
@@ -398,6 +398,50 @@ const std::vector<UCTNode::node_ptr_t>& UCTNode::get_children() const {
     return m_children;
 }
 
+int UCTNode::count_nodes() const {
+    auto nodecount = 0;
+    if (m_has_children) {
+        nodecount += m_children.size();
+        for (auto& child : m_children) {
+            nodecount += child->count_nodes();
+        }
+    }
+    return nodecount;
+}
+
+// Use this version if you know the child is directly under the parent.
+UCTNode::node_ptr_t UCTNode::find_and_take_ownership(const int move) {
+    if (m_has_children) {
+        // TODO: I couldn't get the find_if version to work.
+        // auto& child = std::find_if(begin(m_children), end(m_children),
```

Comment:
find_if probably returns an iterator (behaving more like a ptr, not a ref)

Reply:
Gave up on this, just left it as a loop.